python/ray/data/datasource/file_based_datasource.py

```diff
@@ -607,9 +611,20 @@ def read_files(
                 rows_per_file=self._delegate._rows_per_file(),
                 file_sizes=file_sizes,
             )
-            read_task = ReadTask(
-                lambda read_paths=read_paths: read_files(read_paths, filesystem), meta
-            )
+
+            num_threads = self._delegate.num_threads_per_task()
+
+            def read_task_fn():
+                logger.get_logger().debug(f"Reading {len(read_paths)} files.")
+
+                yield from make_async_gen(
+                    iter(read_paths),
+                    read_files,
+                    num_workers=num_threads,
```

Comment:
can we check the `len(read_paths)` vs `num_threads`, so each thread reads at least one file?